src/Mvc/Mvc.ViewFeatures/src/DefaultHtmlGenerator.cs

```diff
@@ -1329,27 +1347,15 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
 
             case InputType.Text:
             default:
-                var attributeValue = (string)GetModelStateValue(viewContext, fullName, typeof(string));
-                if (attributeValue == null)
+                if (string.Equals(suppliedTypeString, "file", StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(suppliedTypeString, "image", StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: suggest avoiding the empty block e.g.
```suggestion
                // 'value' attribute is not needed for 'file' and 'image' input types.
                if (!string.Equals(suppliedTypeString, "file", StringComparison.OrdinalIgnoreCase) &&
                    !string.Equals(suppliedTypeString, "image", StringComparison.OrdinalIgnoreCase))
                { ... }
```